### PR TITLE
feat: Add configurable send shortcut for chat messages.

### PR DESCRIPTION
**Pull request by Emt-lin**
## issues
- [discord link](https://discord.com/channels/1182847556032659546/1375182675089948764)

## Summary
  Adds user-configurable keyboard shortcuts for sending chat messages, supporting Enter, Shift+Enter, Cmd/Ctrl+Enter, and Alt+Enter options.

  ## Changes

  ### New Features
  - **Configurable Send Shortcut**: Users can now choose their preferred keyboard shortcut for sending messages
  - **4 Shortcut Options**:
    - `Enter` (default)
    - `Shift + Enter`
    - `⌘ + Enter` (macOS) / `Ctrl + Enter` (Windows/Linux)
    - `Alt + Enter`
  - **IME Protection**: Added `isComposing` check to prevent accidental submission during Chinese/Japanese/Korean input
  - **Settings UI**: New dropdown in Basic Settings with help tooltip for hotkey conflicts

  ### Technical Implementation
  - **Settings Migration**: Automatic validation and migration for existing users
  - **Cross-platform Support**: Platform-aware keyboard handling (metaKey vs ctrlKey)
  - **Type Safety**: New `SEND_SHORTCUT` enum in constants

  ## Testing

  ### Test Coverage
  - ✅ **29 unit tests** for KeyboardPlugin
    - All 4 shortcut types
    - Modifier key combinations
    - IME composition handling
    - Cross-platform behavior (macOS vs Windows/Linux)
    - Edge cases
  - ✅ **6 unit tests** for settings migration
    - Missing/invalid value handling
    - Valid value preservation

  ## Breaking Changes
  None - defaults to existing behavior (Enter to send)

  ## Screenshots
Obsidian 2025-10-27 00 43 23

---

> [!NOTE]
> Introduce user-configurable keyboard shortcut to send chat messages (Enter, Shift+Enter, Cmd/Ctrl+Enter, Alt+Enter) with IME-safe handling and settings UI.
> 
> - **Chat Input**:
>   - Update `KeyboardPlugin` to support configurable send shortcuts via `sendShortcut`, add IME composition guard, and platform-aware `Cmd/Ctrl+Enter` handling; export `checkShortcutMatch`.
>   - Wire settings into editor: `LexicalEditor` passes `settings.defaultSendShortcut` to `KeyboardPlugin`.
> - **Settings**:
>   - Add `SEND_SHORTCUT` enum and default `DEFAULT_SETTINGS.defaultSendShortcut`.
>   - Extend `CopilotSettings` with `defaultSendShortcut`; validate/migrate in `sanitizeSettings`.
>   - Add "Send Shortcut" select to `settings/v2/components/BasicSettings.tsx` with help tooltip and macOS/Windows labels.
> - **Tests**:
>   - Add `plugins/KeyboardPlugin.test.ts` covering all shortcut types, modifier combos, IME, and cross-platform logic.
>   - Extend `settings/model.test.ts` to validate `defaultSendShortcut` migration and valid values.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 92c9d103c3fba8438b2730b325623639e38cf077. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by logancyang** [comment]
bugbot run

File: src/components/chat-components/plugins/KeyboardPlugin.tsx [hunk 1]
```diff
@@ -1,39 +1,70 @@
...
+    case SEND_SHORTCUT.SHIFT_ENTER:
+      return event.shiftKey && !event.metaKey && !event.ctrlKey && !event.altKey;
+    case SEND_SHORTCUT.CMD_ENTER:
+      return (Platform.isMacOS ? event.metaKey : event.ctrlKey) && !event.shiftKey && !event.altKey;
+    case SEND_SHORTCUT.ALT_ENTER:
+      return event.altKey && !event.shiftKey && !event.metaKey && !event.ctrlKey;
+    default:
+      return false;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Emt-lin) [reply src/components/chat-components/plugins/KeyboardPlugin.tsx:65]: fixed. cc @logancyang

**Comment by logancyang** [comment]
bugbot run

**Comment by logancyang** [comment]
I tested on mac and cmd + enter doesn't work. Let's keep this simple and only support *Enter* and *Shift + Enter*. @Emt-lin

**Comment by Emt-lin** [comment]
> I tested on mac and cmd + enter doesn't work. Let's keep this simple and only support _Enter_ and _Shift + Enter_. @Emt-lin

see:

CleanShot 2025-10-28 at 18 58 02

**Comment by logancyang** [comment]
My point stands. A lot more explanation for marginal gain at best and more things to be broken at worst. We shouldn't add more for the sake of it. Shift + enter is fine.

**Comment by Emt-lin** [comment]
> My point stands. A lot more explanation for marginal gain at best and more things to be broken at worst. We shouldn't add more for the sake of it. Shift + enter is fine.

agreed. issue: [discord link](https://discord.com/channels/1182847556032659546/1375182675089948764)

**Comment by logancyang** [comment]
> > My point stands. A lot more explanation for marginal gain at best and more things to be broken at worst. We shouldn't add more for the sake of it. Shift + enter is fine.
> 
> agreed. issue: [discord link](https://discord.com/channels/1182847556032659546/1375182675089948764)

I replied there that we will support Shift+Enter without the others.

**Comment by Emt-lin** [comment]
> > > My point stands. A lot more explanation for marginal gain at best and more things to be broken at worst. We shouldn't add more for the sake of it. Shift + enter is fine.
> > 
> > 
> > agreed. issue: [discord link](https://discord.com/channels/1182847556032659546/1375182675089948764)
> 
> I replied there that we will support Shift+Enter without the others.

It's been fixed.
